src/Sync/AsyncFileKeyedMutex.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Amp\File\Sync;
+
+use Amp\File\FilesystemException;
+use Amp\Sync\KeyedMutex;
+use Amp\Sync\Lock;
+use Amp\Sync\SyncException;
+use function Amp\delay;
+use function Amp\File\openFile;
+use function Amp\File\deleteFile;
+
+final class AsyncFileKeyedMutex implements KeyedMutex
```

Comment:
I think we should drop `Async` from the name and name it `KeyedFileMutex`. We should introduce a class alias for the existing implementation without the `Async`prefix as well.

Reply:
I agree, and I think we do not need a class alias, given the new major anyway

Reply:
Good point. I'd still switch the order of words and name it `KeyedFileMutex` instead of `FileKeyedMutex`.

Reply:
True, that's consistent with amphp/sync, so, done.